modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/object_entry/form.jsp

```diff
@@ -51,6 +52,56 @@ portletDisplay.setURLBack(backURL);
 				</clay:row>
 			</clay:sheet-section>
 		</liferay-frontend:fieldset-group>
+
+		<c:if test="<%= objectLayoutBoxCategorization != null %>">
+			<liferay-frontend:fieldset-group>
+				<div class="lfr-form-content" id="<portlet:namespace />categorization">
+					<c:choose>
+						<c:when test="<%= objectEntryDisplayContext.isReadOnly() %>">
+							<liferay-asset:asset-categories-error />
+
+							<liferay-asset:asset-tags-error />
+
+							<aui:fieldset-group markupView="lexicon">
+								<aui:fieldset collapsed="<%= objectLayoutBoxCategorization.isCollapsable() %>" collapsible="<%= objectLayoutBoxCategorization.isCollapsable() %>" label="categorization">
+									<liferay-asset:asset-categories-summary
+										className="<%= objectDefinition.getClassName() %>"
+										classPK="<%= (objectEntry == null) ? 0 : GetterUtil.getLong(objectEntry.getExternalReferenceCode()) %>"
```

Comment:
Hey @seltong, I think you cannot assume that the `externalReferenceCode` is a long always. As far as I know, Camacho is implementing the Salesforce Storage and there it is not a long 😞 . Maybe, this feature must be available for the default storage only.

There is a `getStorageType` method in the `ObjectDefinition` type that you could use to implement this validation.

Thx